engine/access/state_stream/backend/handler_test.go

```diff
@@ -197,6 +199,7 @@ func (s *HandlerTestSuite) TestHeartbeatResponse() {
 	})
 }
 
+// TestExecutionDataStream tests the execution data stream with different event encoding versions.
 func TestExecutionDataStream(t *testing.T) {
```

Comment:
Can you add a test covering `GetExecutionDataByBlockID`?

It looks like we don't have any existing set for that endpoint, so only happy path tests covering both encodings is needed.